core/arch/instr_shared.c

```diff
@@ -2159,6 +2159,37 @@ instr_set_our_mangling(instr_t *instr, bool ours)
         instr->flags &= ~INSTR_OUR_MANGLING;
 }
 
+bool
+instr_is_our_mangling_epilogue(instr_t *instr)
+{
+    ASSERT(!TEST(INSTR_OUR_MANGLING_EPILOGUE, instr->flags) ||
+           instr_is_our_mangling(instr));
+    return TEST(INSTR_OUR_MANGLING_EPILOGUE, instr->flags);
+}
+
+void
+instr_set_our_mangling_epilogue(instr_t *instr, bool epilogue)
+{
+    if (epilogue) {
+        instr->flags |= INSTR_OUR_MANGLING_EPILOGUE;
+    } else
+        instr->flags &= ~INSTR_OUR_MANGLING_EPILOGUE;
+}
+
+instr_t *
+instr_set_translation_mangling_epilogue(dcontext_t *dcontext, instr_t *mangle_instr,
```

Comment:
I found this name mangle_instr confusing: it sounds like one or our inserted instructions for mangling, but in usage this is the app instruction.  Maybe it's better to remove this parameter and instead call instrlist_get_translation_target() here?  That will avoid subtle bugs with "instr" being something else: we know the instrlist xl8 is the right thing.

Reply:
And pass in ilist instead?

Reply:
That would be necessary to call instrlist_get_translation_target(), yes.

Reply:
Done